frontend2/src/components/tables/chart/TeamChart.tsx

```diff
@@ -14,28 +19,62 @@ type UTCMilliTimestamp = number;
 // { "Gone Sharkin" : [ [ 1673826806000.0, 1000 ], [1673826805999.0, 900], ... ], ...}
 export interface TeamChartProps {
   yAxisLabel: string;
-  values: Record<string, Array<[UTCMilliTimestamp, number]>>;
+  values?: Record<string, ChartData[]>;
+  loading?: boolean;
+  loadingMessage?: string;
 }
 
-const TeamChart = ({ yAxisLabel, values }: TeamChartProps): JSX.Element => {
-  const seriesData: Highcharts.SeriesOptionsType[] = [];
-
+const TeamChart: React.FC<TeamChartProps> = ({
+  yAxisLabel,
+  values,
+  loading = false,
+  loadingMessage,
+}) => {
   // Translate values into Highcharts compatible options
-  for (const team in values) {
-    const formattedEntry: Highcharts.SeriesOptionsType = {
+  const [myChart, setChart] = useState<Highcharts.Chart>();
+
+  const seriesData: Highcharts.SeriesOptionsType[] = useMemo(() => {
+    if (values === undefined) return [];
+    return Object.keys(values).map((team) => ({
       type: "line",
       name: team,
       data: values[team],
       marker: {
         enabled: false,
         symbol: "circle",
       },
-    };
-    seriesData.push(formattedEntry);
+    }));
+  }, [values]);
+
+  if (myChart !== undefined) {
+    try {
+      if (loading) myChart.showLoading(loadingMessage ?? "Loading...");
+      else myChart.hideLoading();
+    } catch (e) {
+      // Ignore internal highcharts errors...
+    }
```

Comment:
I think the reason there is an error here is because myChart.options is not defined. Could we define it before we call showLoading?

![image](https://github.com/battlecode/galaxy/assets/40174697/6c944fce-8d16-42d8-9733-52e69ed5f1e9)

Reply:
Ok trying to do that causes other errors nvm 💀

Reply:
actually i think we may just need to put it in a useeffect. 
![image](https://github.com/battlecode/galaxy/assets/40174697/16fea33d-f985-453b-8539-ee156c6ca64e)